Remotion/Data/DomainObjects/Queries/Configuration/QueryDefinition.cs

```diff
@@ -269,6 +286,7 @@ protected virtual void GetObjectData (SerializationInfo info, StreamingContext c
       info.AddValue("Statement", _statement);
       info.AddValue("QueryType", _queryType);
       info.AddValue("CollectionType", _collectionType);
+      info.AddValue("Metadata", _metadata);
```

Comment:
Since we can have any type of `object` in our metadata, what happens if one of our values is not serializable?

Reply:
I have no clue, when using metadata correctly, it should not have any non serializable things inside (mostly strings or rather enums to cover the metadata aspect.)
But other than that, I'm not sure how this should behave.
@MichaelKetting?

Reply:
@Frazerus & @martin-herwich : if  a value is not serializable, the serialization will fail. For enums and strings as values, that's a non-issue. If someone decides to put complex datastructures into the metadata, all bets are off.

For the backport, the projects don't use serialization. For re-motion v7, we will be dropping serialization anyway. For our own default value (an empty readonly dictionary, we should make sure the type is serializable. 

Perhaps it would be a good idea/helpful to note in the xmldoc for the parameter that the metadata dictionary and it's values must be serializable if the project uses serialization.